dvc/repo/list_outs.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import os
+
+
+logger = logging.getLogger(__name__)
+
+
+@staticmethod
+def list_outs(url, targets=[], rev=None, recursive=None):
+    if os.path.exists(url):
+        path = os.path.abspath(url)
+        return _list_outs_local_repo(path, targets, recursive)
+    else:
+        return _list_outs_git_repo(url, targets, rev, recursive)
+
+
+def _list_outs_local_repo(path, targets=[], recursive=None):
+    from dvc.repo import Repo
+    from dvc.path_info import PathInfo
+
+    repo = Repo(path)
+
+    outs = set()
+    for stage in repo.stages:
+        outs = outs.union(stage.outs)
```

Comment:
Looks like we don't currently support git files. We should list them too along with dvc-outputs.

Reply:
So you expect that `dvc list` list regular files tracked by git?
```
dvc list git@github.com:repo
.gitignore
data/data.xml
```
If I got right then I think it's not a right way cause
1. It's not clear user usage (motives) of the command. Right now we are listing artifacts for getting them, for working with regular files there are `git` commands
2. Needs to differentiate somehow dvc artifacts from regular files, and add complexity for pipelining commands
    ```
    git list git@github.com:repo
    .gitignore
    outs:data/data.xml
    ...
    git list git@github.com:repo | grep outs: | awk ..
    ```
3. It makes command slower then it is, especially on huge repos

Reply:
1. `dvc import` and `dvc get` work with git files just fine, without any additional flags, so it makes sense to make `dvc list` behave that same way. :slightly_smiling_face: 

2. Differentiation is not required, as `dvc get/import` work the same with git-tracked and dvc-tracked files. :slightly_smiling_face: 

3. Because of us having to list git files too?

Reply:
> `dvc import` and `dvc get` work with git files just

ah, right, now it makes sense

> Differentiation is not required

Ok, maybe it can be shown in future with `dvc list -l repo`

> Because of us having to list git files too?

Yes, mostly time I've spent is trying to speedup the process of listing (skip history, not necessary files, etc). Fetching whole repo before listing doesn't look good for me. But I can retrieve `dvc-files` only and list all other files through `git ls-tree` without fetching them
I think in that case it can be fast enough

Reply:
**Proposal**
Maybe needs to support filter option for `list` as `--outputs` for showing outputs only?
```
dvc list --outputs git@github.com:repo
```

Reply:
@JIoJIaJIu I would probably not focus on that kind of optimization for now and would use already available mechanisms instead. We have some clone caching present with `git pull`s on top, which is quite fast and we might simply consider making that cache persist over multiple `dvc` invocations instead of messing around with sparse clone/checkout. That being said, I haven't looked into the sparse clone implementation deep enough yet, maybe it would be reasonable to keep, not sure.